ngraph/core/src/op/transpose.cpp

```diff
@@ -50,14 +50,21 @@ void op::v1::Transpose::validate_and_infer_types()
 
     const auto& arg_shape = get_input_partial_shape(0);
     NODE_VALIDATION_CHECK(this,
-                          input_order_shape.compatible(PartialShape{arg_shape.rank()}),
+                          input_order_shape.compatible(PartialShape{arg_shape.rank()}) ||
+                              (input_order_shape.is_static() && input_order_shape.rank() == 1 &&
```

Comment:
Why do you added this check for `input_order_shape[0] == 0`? I do understand that if the input tensor rank is 1 then it is the only correct permutation. But if you want to check a permutation then it should be done separately and for all valid input ranks

Reply:
The problem that in order to create a empty constant we need to create something like that:
```
Constant(Shape{0}, {} /*data*/)
```
This order shape is incompatible with data shape, but we know that it is a special case when we need to reverse input shapes order. Exactly this case I checked here.